internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -183,6 +183,21 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 			nextCheckTimes = append(nextCheckTimes, nextCheck)
 		}
 	}
+
+	// check certificate expiration
+	if t.MHC.Spec.Certificates != nil && t.MHC.Spec.Certificates.ExpiresWithinDays != nil {
+		// Check only if the certificate expiry information is available.
+		if t.Machine.Status.CertificatesExpiryDate != nil {
+			certificatesExpireAt := t.Machine.Status.CertificatesExpiryDate.Time
+			if now.Add(time.Duration(*t.MHC.Spec.Certificates.ExpiresWithinDays) * 24 * time.Hour).After(certificatesExpireAt) {
+				expiresIn := certificatesExpireAt.Sub(now)
+				expiredInRoundedToMinutes := expiresIn.Round(time.Minute)
```

Comment:
I'm not sure if minutes is the right granularity. I'm not sure how often we run into the MHC reconcile, but this would give us a different condition message every minute. I.e. potentially one update on a Machine every minute.